server/swap/swap.go

```diff
@@ -164,8 +168,9 @@ type Swapper struct {
 	// authMgr is an AuthManager for client messaging and authentication.
 	authMgr AuthManager
 	// The matches map and the contained matches are protected by the matchMtx.
-	matchMtx sync.RWMutex
-	matches  map[string]*matchTracker
+	matchMtx     sync.RWMutex
+	matches      map[string]*matchTracker
+	orderMatches map[order.OrderID]int
```

Comment:
Referring to the commit message for https://github.com/decred/dcrdex/pull/206/commits/5042449a0831c8e2c4942c19f2f7c9cb4d914624:

> TODO: there is a possible race in the window between when Matcher
(in Market) reduces an order's remaining amount and when Negotiate
registers the new active swap for the order. This could result in an order
being flagged as complete by processAck when really the final swap is
about to begin.  To resolve this, a lock in Market.processReadyEpoch
would be required to make Match and Negotiate atomic with respect
to order fill/remaining amounts and active swaps.

This is where a separate mutex locking `orderMatches` may be warranted, which would also solve the issue mentioned in the above comment via new exported `Swapper` methods just for holding/releasing the lock.